cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/namespace/NamespaceLister.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.master.spi.namespace;
+
+import java.util.List;
+
+/**
+ * Interface for listing details of CDAP namespaces.
+ */
+public interface NamespaceLister {
+  /**
+   * Provides details about all available namespaces.
+   */
+  void listNamespaces(List<NamespaceDetail> namespaceDetails);
```

Comment:
A verb like listNamespaces() sounds like this method should return a list. I think something like onStart() or initializeNamespaces() would be better.

I think it also makes sense for this to have a default no-op implementation.

It's also more generic to pass in a Collection rather than a List

Reply:
The javadoc should also describe the contract in more detail.

For example, this probably needs to be an idempotent operation, as we'll want to retry it if it throws an exception. 

Do we need to differentiate between exception that are retried vs those that should just fail and move on? Or should we retry everything? (It seems like we can retry repeatedly, but haven't thought through everything)

Reply:
done